### PR TITLE
[CodeCamp2023-526] Kullback-Leibler divergence Loss implementation

### PR DESCRIPTION
**Pull request by zen0no**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

It's OpenMMLab  Codecamp task.

## Modification

Implementd  Kullback-Leibler divergence loss and also added tests for it.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMDet3D.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

**Comment by xiexinch** [comment]
Hi @zen0no,
Thanks for your contribution! 
Please fix the lint error.

**Comment by crazysteeaam** [comment]
Hi @zen0no,
Could you please prefix the PR title with [CodeCamp2023-526]? Thank you!

**Comment by zen0no** [comment]
Hi @crazysteeaam,
I have a problem with lint. When I check all hooks with pre-commit on my local machine, it passes, but it fails on yapf test in  circleci

Link for the problem in circleci: https://app.circleci.com/pipelines/github/open-mmlab/mmsegmentation/1751/workflows/0f284d64-9863-477b-a1c7-569668b4aa31/jobs/5655?invite=true#step-103-46

**Comment by OpenMMLab-Assistant-004** [comment]
Hi @zen0no,

We'd like to express our appreciation for your valuable contributions to the mmsegmentation. Your efforts have significantly aided in enhancing the project's quality.
It is our pleasure to invite you to join our community thorugh Discord_Special Interest Group (SIG) channel. This is a great place to share your experiences, discuss ideas, and connect with other like-minded people. To become a part of the SIG channel, send a message to the moderator, OpenMMLab, briefly introduce yourself and mention your open-source contributions in the #introductions channel. Our team will gladly facilitate your entry. We eagerly await your presence. Please follow this link to join us: ​https://discord.gg/UjgXkPWNqA.

If you're on WeChat, we'd also love for you to join our community there. Just add our assistant using the WeChat ID: openmmlabwx. When sending the friend request, remember to include the remark "mmsig + Github ID".

Thanks again for your awesome contribution, and we're excited to have you as part of our community!

File: mmseg/models/losses/kldiv_loss.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        assert input.shape == target.shape, 'Input and target ' \
+            'must have same shape,' \
+            f'but got shapes {input.shape} and {target.shape}'
+
+        input = F.softmax(input / self.temperature, dim=1)
+        target = F.softmax(target / self.temperature, dim=1)
+
+        loss = F.kl_div(input, target, reduction='none', log_target=False)
+        loss = loss * self.temperature * self.temperature
+
```
Comment (xiexinch) [inline mmseg/models/losses/kldiv_loss.py:68]:
```suggestion
        loss = loss * self.temperature**2

```

**Review by xiexinch — APPROVED** [review]

File: mmseg/models/losses/kldiv_loss.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import torch.nn as nn
+import torch.nn.functional as F
+
+from mmseg.registry import MODELS
+
+
+@MODELS.register_module()
+class KLDivLoss(nn.Module):
+
+    def __init__(self, temperature=1.0, reduction='mean'):
```
Comment (xiexinch) [inline mmseg/models/losses/kldiv_loss.py:12]: We might add type hints.
